server/src/main/java/io/crate/metadata/IndexReference.java

```diff
@@ -101,6 +101,18 @@ public IndexReference(int position,
         this.analyzer = analyzer;
     }
 
+    public IndexReference(int position,
```

Comment:
as said, `IndexReference` is not losing nullable/docValues flag in `DocIndexMetadata` but I would need such ctor for add column. Can do it in add column PR if it makes more sense. The only reason I added it here was kinda clustering similar changes together.

Reply:
It's easier to reason about the changes if this is done in the other PR. If you look at the diff at a later point it won't be easily visible anymore that this change here is completely unrelated to the fix.

Reply:
dropped this commit, thanks for the feedback!